pkg/apis/management.cattle.io/v3/cluster_types.go

```diff
@@ -354,6 +358,10 @@ type SaveAsTemplateOutput struct {
 	ClusterTemplateRevisionName string `json:"clusterTemplateRevisionName,omitempty"`
 }
 
+type AKSStatus struct {
+	UpstreamSpec          *aksv1.AKSClusterConfigSpec `json:"upstreamSpec"`
```

Comment:
Could be done in another PR after some more investigation but I'm guessing we will need something like `PrivateRequiresTunnel` like GKEStatus and EKSStatus do.

Reply:
ok, added